### ISSUE TITLE
[QUESTION] JDK or JRE?

### ISSUE THREAD

**Issue by RoiArthurB** [gama-platform/gama#2889]
Hello,

I want to create a docker image of GAMA (which should not be too complicated... Or at least I hope so!) and I'm listing everything which has to be bundled in the container.

I'm looking for GAMA-friendly Linux JDK in a container, and it looks that mainly people upload JRE in place of JDK. The logic behind is that a container will only execute java application and not compile them and, so, a JRE is enough.

So, I'm wondering if GAMA needs a full JDK to be executed or if a simple JRE will be enough? The container will (I think) mainly be used for the headless aspect, so maybe I don't need a JDK inside... Or does GAMA need it to compile GAML files and start the simulation? 🤔🤔🤔

**Comment by hqnghi88** [comment]
Hi,
Docker of gama git with eclipse and java or docker of gama? If gams git,
then jdk , yes, we must do compilation. Docker of gama for what??? An
installer?

**Comment by RoiArthurB** [comment]
For now, I just wanted to start with the GAMA v1.8.0 (to have a fix & stable container). My point is to set up a GAMA headless easily on a server (like on Digital Ocean with a docker droplet and just having to pull GAMA on the official Docker Hub and starting from there). 

And I think that GAMA could only be used on the headless mode from Docker, because its core doesn't support any GUI process, you can't display any window from a container (as far as I know at least)

So do you think that a JRE could be enough? The container would be lighter if so, otherwise I'll go with the full OpenJDK (easier to setup from a Linux base and I never ran in any problem using it)

**Comment by hqnghi88** [comment]
Yes, it is very good to have a docker config. Its a good beginning for gama
git docker also. I must say that for some reason, i have seen some gama did
not work well with jre on some machine. So we chose to embed whole jdk. But
if you have such knowledge about jdk jre or openjdk,, you can try with jre,
it should work as you said gama dont need to be recompiled

**Comment by RoiArthurB** [comment]
And does GAMA need whole JDK to compile GAML files and experiment or a simple JRE should do the trick?

Also, to have a proper GAMA container etc, it would be nice to have a Gama-platform official account on Docker Hub, @AlexisDrogoul do you want me to create it or I let you make it and I'll wait for credentials to push containers (when it will be working) on it? I'm not sure, but it looks as complicated as creating a Github account and pushing on a repo (short tutorial [here](https://dev.to/ech0server/creating-a-docker-image-and-upload-it-to-docker-hub) )
